str_analysis/check_combined_results_tsv_for_pathogenic_repeats.py

```diff
@@ -0,0 +1,274 @@
+        "combine_expansion_hunter_json_to_tsv.py script. It's assumed that combine_expansion_hunter_json_to_tsv.py "
+        "was run with --sample-metadata and --variant-catalog args to add sample-level and locus-level metadata columns")
+
+    # Parse and validate command-line args + read in the combined table(s) from the given command_tsv_path(s)
+    args = p.parse_args()
+
+    all_dfs = []
+    all_locus_ids = set()
+    for combined_tsv_path in args.combined_tsv_path:
+        if not os.path.isfile(combined_tsv_path):
+            p.error(f"{combined_tsv_path} not found")
+
+        df = pd.read_table(combined_tsv_path, low_memory=False)
+
+        # check that all required columns are present
+        missing_required_columns = set(REQUIRED_COLUMNS) - set(df.columns)
+        if missing_required_columns:
+            p.error(f"{combined_tsv_path} is missing these required columns: {missing_required_columns}")
+
+        # fill in values for missing optional columns
+        missing_optional_columns = set(OPTIONAL_COLUMNS) - set(df.columns)
+        if missing_optional_columns:
+            print(f"WARNING: {combined_tsv_path} is missing these columns: {missing_optional_columns}. "
+                  f"Filling them with None...")
+            for c in missing_optional_columns:
+                df.loc[:, c] = None
+
+        all_locus_ids |= set(df.LocusId)
+        all_dfs.append((df, combined_tsv_path))
+
+    # Process each locus
+    for locus_id in sorted(all_locus_ids):
+        if args.locus and locus_id not in args.locus:
+            continue
+
+        print(">"*100)  # print a divider
+        for i, (full_df, df_source_path) in enumerate(all_dfs):
+
+            print("="*100)  # print a divider
+            print(f"** {locus_id} from {df_source_path} **")
+
+            locus_specific_df = full_df[full_df.LocusId == locus_id]
+            if len(locus_specific_df) == 0:
+                continue
+
+            locus_specific_df = locus_specific_df.sort_values(by=["Num Repeats: Allele 2", "Num Repeats: Allele 1"], ascending=False)
+
+            # get the 1st row and use it to look up metadata values which are the same across all rows for the locus (eg. Inheritance Mode)
+            first_row = locus_specific_df.iloc[0].to_dict()
+            inheritance_mode = first_row.get("VariantCatalog_Inheritance") or "Unknown"
+
+            disease_info = first_row.get("VariantCatalog_Diseases")
+            intermediate_threshold_min = None
+            pathogenic_threshold_min = None
+            if disease_info:
+                try:
+                    disease_info = literal_eval(disease_info)
+                except Exception as e:
+                    raise ValueError(f"Unable to parse {disease_info} as json: {e}")
+
+                try:
+                    intermediate_threshold_min = min(int(float(d["NormalMax"]) + 1) for d in disease_info if d.get("NormalMax"))
```

Comment:
I think I just used low_memory to get rid of the pandas warning. The script should work regardless. 
The conversion from float was because I was seeing values like "25.0" in some cases. I think this line would fail if d["NormalMax"] was NaN (when converting to int). I should add better check though.